scico/functional/_functional.py

```diff
@@ -52,6 +52,13 @@ def __mul__(self, other):
     def __rmul__(self, other):
         return self.__mul__(other)
 
+    def __add__(self, other):
+        if isinstance(other, Functional):
+            return FunctionalSum(self, other)
+        raise NotImplementedError(
+            f"Operation __add__ not defined between {type(self)} and {type(other)}"
```

Comment:
The recommended way to do this is to return the constant `NotImplemented` rather than raise an exception; see https://docs.python.org/3/library/constants.html#NotImplemented

Reply:
Fixed. (I wouldn't be surprised if there are some other instances of this error floating around in other modules.)